packages/eslint-plugin/configs/recommended.js

```diff
@@ -25,6 +25,12 @@ if ( isPackageInstalled( 'prettier' ) ) {
 	const prettierConfig = { ...defaultPrettierConfig, ...localPrettierConfig };
 	config.rules = {
 		'prettier/prettier': [ 'error', prettierConfig ],
+		// Prettier _disables_ this rule, but we want it!
+		// See https://github.com/prettier/eslint-config-prettier?tab=readme-ov-file#curly
+		// > This rule requires certain options.
+		// > …
+		// > If you like this rule, it can be used just fine with Prettier as long as you don’t use the "multi-line" or "multi-or-nest" option.
+		curly: [ 'error', 'all' ],
```

Comment:
It was likely a conscious decision to let Prettier format everything and remove all the opinions from ESLint. More context:
https://make.wordpress.org/core/2019/12/09/proposed-javascript-coding-standards-revisions-for-prettier-compatibility/

It's probably fine to change it for Gutenberg, but from the standpoint of WordPress projects that would be a breaking change that would require communicating with the community on Make Core blog.

Reply:
Yep, I'll make this a breaking change.

I'm digging in history, but I suspect disabling this rule when prettier is enabled was an oversight.

Reply:
All the prettier eslint changes started in https://github.com/WordPress/gutenberg/pull/18048 and I believe `curly` has been effectively disabled since then. I don't see discussion of this particular point so I'm not sure whether it was intentional or not.

As far as I can tell, we intended to remove the eslint rules that explicitly conflict with prettier, and `curly` _may_ conflict but never did in our case, and it was disabled without considering it specifically.

Reply:
`curly` is not a formatting option and thus not really something Prettier covers, so it's not really a conflicting rule (unless the multi-line option is chosen). IMO it makes sense to explicitly enable it.